libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -240,6 +240,7 @@ void ESP8266WiFiGenericClass::_eventCallback(void* arg)
 
     if (event->event == EVENT_STAMODE_AUTHMODE_CHANGE) {
         auto& src = event->event_info.auth_change;
+        WiFi.authMode = src.new_mode;
```

Comment:
Is this is a side-effect of attempted authmode (i.e. wpa3 downgrade to wpa2) or event that always happen every connection?

Reply:
after every connection. 
src.old_mode is 0